airbrakes/mock/mock_imu.py

```diff
@@ -34,22 +36,59 @@ def __init__(self, log_file_path: Path, real_time_simulation: bool):
         # is more realistic for a real flight.
         if not real_time_simulation:
             self._data_queue: multiprocessing.Queue[IMUDataPacket] = multiprocessing.Queue(15)
+        self._data_fetch_process = multiprocessing.Process(
+            target=self._fetch_data_loop,
+            args=(
+                log_file_path,
+                real_time_simulation,
+                start_after_log_buffer,
+            ),
+            name="Mock IMU Process",
+        )
 
-    def _fetch_data_loop(self, log_file_path: Path, real_time_simulation: bool) -> None:
+    def _fetch_data_loop(
+        self, log_file_path: Path, real_time_simulation: bool, start_after_log_buffer: bool = False
+    ) -> None:
         """
         Reads the data from the log file and puts it into the shared queue.
         :param log_file_name: the name of the log file to read data from located in logs/
         :param real_time_simulation: Whether to simulate a real flight by sleeping for a set period, or run at full
             speed, e.g. for using it in the CI.
+        :param start_after_log_buffer: Whether to send the data packets only after the log buffer
+            was filled for Standby state.
         """
         signal.signal(signal.SIGINT, signal.SIG_IGN)  # Ignore the KeyboardInterrupt signal
 
+        start_index = 0
+        if start_after_log_buffer:
+            # We need to loop through the file and calculate the time difference to find the index
+            # at which the log buffer starts. That index will be used as the start point.
+            with log_file_path.open(newline="") as csvfile:
+                reader = csv.DictReader(csvfile)
+
+                # Reads each row as a dictionary
+                row: dict[str, str]
+                init_timestamp = None
+                for idx, row in enumerate(reader):
+                    timestamp = convert_to_nanoseconds(row["timestamp"])
+                    if init_timestamp is None:
+                        init_timestamp = timestamp
+                        continue
+                    if timestamp - init_timestamp > 1e9:  # Anything greater than 1 second is end of the buffer.
+                        start_index = idx
+                        break
+                    init_timestamp = timestamp
+
         with log_file_path.open(newline="") as csvfile:
             reader = csv.DictReader(csvfile)
 
             # Reads each row as a dictionary
             row: dict[str, str]
-            for row in reader:
+            for idx, row in enumerate(reader):
+                # If we are starting after the log buffer, skip the rows before the buffer:
```

Comment:
Can you make it so it doesn't completely skip StandByState? Like maybe it a second before the buffer ends, or smth like that?

Reply:
it does not completely skip Standbystate. The log buffer also writes 5000 packets before Motorburn, which is Standbystate

Reply:
Oh so it only skips the first buffer

Reply:
yep